src/Edge.java

```diff
@@ -1,12 +1,15 @@
+import java.util.ArrayList;
+import java.util.HashSet;
+
 public class Edge {
 
-    private Node reference;
+    private ArrayList<Node> references = new ArrayList<>();
```

Comment:
No need for a list here. An edge is what connects two nodes. Given that our graph is directed, there's no need to keep a pointer to the starting node. So this becomes a wrapper for a pointer towards a node. We can even do without it for now, but we'll need it when we get to the associativity part and have to assign weight to each edge.